Compilation error due to ExecutorService API change in JDK 19

### Version

3b16da43563abcfa1225bae1c4ddcf862acf9912

### Bug description

Starting in JDK 19, `ExecutorService` implements `AutoCloseable`: https://github.com/openjdk/jdk/commit/00e6c63cd12e3f92d0c1d007aab4f74915616ffb#diff-139913957207a1eeaa338c35b269f2c566b358590ae62dd2a8063e98bb5b0456

This change causes a compilation error in mina-sshd.

### Actual behavior

```
cd ~/sshd-common
mvn clean package -Djava.sdk.version=19
...
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project sshd-common: Compilation failure: Compilation failure:
[ERROR] /usr/local/google/home/cushon/src/mina-sshd/sshd-common/src/main/java/org/apache/sshd/common/util/threads/NoCloseExecutor.java:[43,8] org.apache.sshd.common.util.threads.NoCloseExecutor is not abstract and does not override abstract method close() in java.nio.channels.Channel
[ERROR] /usr/local/google/home/cushon/src/mina-sshd/sshd-common/src/main/java/org/apache/sshd/common/util/threads/CloseableExecutorService.java:[29,8] types java.util.concurrent.ExecutorService and org.apache.sshd.common.Closeable are incompatible;
[ERROR]   interface org.apache.sshd.common.util.threads.CloseableExecutorService inherits unrelated defaults for close() from types java.util.concurrent.ExecutorService and org.apache.sshd.common.Closeable
[ERROR] /usr/local/google/home/cushon/src/mina-sshd/sshd-common/src/main/java/org/apache/sshd/common/util/threads/SshThreadPoolExecutor.java:[35,8] types org.apache.sshd.common.Closeable and java.util.concurrent.ExecutorService are incompatible;
[ERROR]   class org.apache.sshd.common.util.threads.SshThreadPoolExecutor inherits unrelated defaults for close() from types org.apache.sshd.common.Closeable and java.util.concurrent.ExecutorService
[ERROR] -> [Help 1]
```

### Expected behavior

I expected the project to compile

### Relevant log output

```Shell
Apache Maven 3.8.6
Maven home: /usr/share/maven
Java version: 19.0.1, vendor: Oracle Corporation
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", arch: "amd64", family: "unix"
```


### Other information

_No response_

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -26,6 +26,7 @@
 
 * [GH-268](https://github.com/apache/mina-sshd/issues/268) (Regression in 2.9.0) Heartbeat should throw an exception if no reply arrives within the timeout.
 * [GH-275](https://github.com/apache/mina-sshd/issues/275) SFTP: be more lenient when reading SSH_FXP_STATUS replies.
+* [GH-285](https://github.com/apache/mina-sshd/issues/285) Fix compilation failure on Java 19.
 
 ## Major code re-factoring
 

--- a/sshd-common/src/main/java/org/apache/sshd/common/util/threads/CloseableExecutorService.java
+++ b/sshd-common/src/main/java/org/apache/sshd/common/util/threads/CloseableExecutorService.java
@@ -19,6 +19,7 @@
 
 package org.apache.sshd.common.util.threads;
 
+import java.io.IOException;
 import java.time.Duration;
 import java.util.Objects;
 import java.util.concurrent.ExecutorService;
@@ -30,5 +31,10 @@ public interface CloseableExecutorService extends ExecutorService, Closeable {
     default boolean awaitTermination(Duration timeout) throws InterruptedException {
         Objects.requireNonNull(timeout, "No timeout specified");
         return awaitTermination(timeout.toMillis(), TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    default void close() throws IOException {
+        Closeable.super.close();
     }
 }

